### PR TITLE
Add simple scripts to check perm on shell files

### LINKED ISSUE #9417: Prow: lint shell file mode
Shell scripts should be marked executable... 🤦‍♂️

### PR DESCRIPTION
**Pull request by matthyx**
@BenTheElder this is probably not what you requested, but I mean, why reinvent the wheel?

The other idea of creating a small utility, packaging it in a docker image, etc... seems also good to me, but I failed to see where to put the sources?

fixes #9417

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
The draw of the utils was to not need to package stuff up, etc

**Comment by BenTheElder** [comment]
This seems like a great start, if it stays this small repos can just copy the script and the job, the idea behind packaging it in an image (if it gets more complex) is that we can manage building the image and then other repos just schedule a job with that image + command and don't need to do anything else to setup.

**Comment by matthyx** [comment]
As you wish guys, but packaging just one shell script seems overkill...
What would make sense would be to package the whole hack scripts into an image that could as well include bazel and all utilities needed to build and verify the project.
Maybe something to discuss during SIG meeting?

**Comment by stevekuznetsov** [comment]
Common tooling (like Go) for static analysis sounds like a cool goal in the future. @BenTheElder should we add this job as-is for now?

**Comment by BenTheElder** [comment]
Yes, and +1 on it being overkill for now, but these things always grow so noting for the future.

The other issue with just copying the script around is that usually nobody ever copies the fixes around :^)

File: hack/update-file-perms.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+find . -name "*.sh" \! -perm /a+x | xargs chmod +x
```
Comment (BenTheElder) [inline hack/update-file-perms.sh]:
```console
Line 20:
find . -name "*.sh" \! -perm /a+x | xargs chmod +x
^-- SC2038: Use -print0/-0 or -exec + to allow for non-alphanumeric filenames.
```

Comment (BenTheElder) [reply hack/update-file-perms.sh]:
```suggestion
find . -name "*.sh" \! -perm /a+x -exec chmod +x {} +
```

Comment (matthyx) [reply hack/update-file-perms.sh]: Never thought about non-alpha chars... thx for the tip!

**Comment by matthyx** [comment]
Should we prepare a KEP for the dockerized tooling?
Or will you add that to your backlog somewhere?

**Comment by BenTheElder** [comment]
@matthyx I don't think we need a kep, our tools are defacto containerized and hopefully we don't fully implement these from scratch. we may even be able to just recommend existing linter images.

File: hack/update-file-perms.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+find . -name "*.sh" \! -perm /a+x -exec chmod +x {}
```
Comment (BenTheElder) [inline hack/update-file-perms.sh]: this needs the trailing `+`

Comment (BenTheElder) [reply hack/update-file-perms.sh]: see: http://man7.org/linux/man-pages/man1/find.1.html

Comment (BenTheElder) [reply hack/update-file-perms.sh]:
```suggestion
find . -name "*.sh" \! -perm /a+x -exec chmod +x {} +
```

Comment (matthyx) [reply hack/update-file-perms.sh]: oops

**Comment by BenTheElder** [comment]
Thanks!

**Comment by krzyzacy** [comment]
... start with `optional: true` or only run if a .sh file changes?

**Pull request merged**